addons/api/tests/unit/utils/indexed-db-query-test.js

```diff
@@ -0,0 +1,521 @@
+        },
+        scope_id: {
+          logicalOperator: 'and',
+          values: [{ equals: scope2.id }, { notEquals: scope1.id }],
+        },
+      },
+    };
+
+    const result = await queryIndexedDb(indexedDb.db, resource, query);
+    assert.strictEqual(result.length, 5);
+  });
+
+  test('it cannot filter with "contains" as initial operation', async function (assert) {
+    assert.expect(1);
+    await seedIndexDb(resource, store, indexedDb, this.server);
+    const query = {
+      filters: {
+        scope_id: [{ contains: scope1.id }],
+      },
+    };
+
+    try {
+      await queryIndexedDb(indexedDb.db, resource, query);
+    } catch (e) {
+      assert.strictEqual(
+        e.message,
+        'Contains is not supported as the first filter.',
+      );
+    }
+  });
+
+  test('it can filter with "contains" as subsequent operation', async function (assert) {
+    assert.expect(1);
+    await seedIndexDb(resource, store, indexedDb, this.server);
+    const query = {
+      filters: {
+        scope_id: {
+          logicalOperator: 'and',
+          values: [{ equals: scope1.id }, { contains: scope2.id.slice(0, 2) }],
+        },
+      },
+    };
+
+    const result = await queryIndexedDb(indexedDb.db, resource, query);
+    assert.strictEqual(result.length, 5);
+  });
+
+  test('it filters by single field with "gt" as initial operation', async function (assert) {
+    await seedIndexDb(resource, store, indexedDb, this.server);
+    const query = {
+      filters: {
+        scope_id: {
+          logicalOperator: 'and',
+          values: [{ gt: scope1.id }, { equals: scope2.id }],
+        },
+      },
+    };
+
+    let result = await queryIndexedDb(indexedDb.db, resource, query);
+
+    assert.strictEqual(result.length, 5);
+
+    query.filters.scope_id.logicalOperator = 'or';
+
+    result = await queryIndexedDb(indexedDb.db, resource, query);
+
+    assert.strictEqual(result.length, 5);
+  });
+
+  test('it filters by single field with "gt" as subsequent operation', async function (assert) {
+    await seedIndexDb(resource, store, indexedDb, this.server);
+    const query = {
+      filters: {
+        scope_id: {
+          logicalOperator: 'and',
+          values: [{ equals: scope2.id }, { gt: scope1.id }],
+        },
+      },
+    };
+
+    let result = await queryIndexedDb(indexedDb.db, resource, query);
+
+    assert.strictEqual(result.length, 5);
+
+    query.filters.scope_id.logicalOperator = 'or';
+
+    result = await queryIndexedDb(indexedDb.db, resource, query);
+
+    assert.strictEqual(result.length, 5);
+  });
+
+  test('it filters by single field with "gte" as initial operation', async function (assert) {
+    await seedIndexDb(resource, store, indexedDb, this.server);
+    const query = {
+      filters: {
+        scope_id: {
+          logicalOperator: 'and',
+          values: [{ gte: scope1.id }, { equals: scope2.id }],
```

Comment:
What are the expected results when using comparison operators on IDs? Aren't they strings?

Reply:
Yeah they are strings scope1.id has the value of `p_123` and scope2.id has the value of `p_456` so scope1.id will always be less than scope2.id. I can remove that second operator since it's not necessary, I thought we should test multiple operations on a single filter. However, that was before adding the other test cases where the operation is second and not first.

Reply:
Right but I suppose my question is should we be comparing strings since that's what the IDs just are, can we use fields that are actually numbers? I don't want us to be accidentally testing something that's like `"2" > "10" === true`

Reply:
Currently, none of the modelIndexes have an index that is a number type. Do we want to add an additional index field or add another resource that will be indexed on a number field ?

Reply:
I was thinking of just adding another resource, we can just add a TODO to come back to this if you'd like to wait

Reply:
yeah i can come back and update these since I'm not sure what fields we are indexing on for the other resources.